### ISSUE TITLE
Local dev workflow:  DirectMigration saying 'ConfigMap migration-controller' not found.  MigMigration is waiting

### ISSUE THREAD

**Issue by jwmatthews** [konveyor/mig-controller#918]
I  hit the below when attempting a Direct Migration.
Steps:
1) Install 1.3.2 and create MigClusters
2) Follow [Hacking guide ](https://github.com/konveyor/mig-controller/blob/master/HACKING.md) to install latest controller source and run locally
3) Use the MigClusters previously created
4) Create a new Plan
5) Do a Direct Migration.

Is this a bug with upgrades from 1.3.2 to 1.4.0?
Also concerned that the UI did not show an error and thinks this is still running.
I am running from master, possible the error is self-inflicted from my environment.

```
{"level":"info","ts":1611442849.2483509,"logger":"direct|z9pwb","msg":"","direct":"openshift-migration/0e44ae00-5dc3-11eb-9ca8-df7e5254778b-7qtx8","error":"ConfigMap \"migration-controller\" not found","stacktrace":"\ngithub.com/konveyor/mig-controller/pkg/controller/directvolumemigration.(*Task).Run()\n\t/Users/jmatthews/git/jwmatthews/mig-controller/pkg/controller/directvolumemigration/task.go:314\ngithub.com/konveyor/mig-controller/pkg/controller/directvolumemigration.(*ReconcileDirectVolumeMigration).migrate()\n\t/Users/jmatthews/git/jwmatthews/mig-controller/pkg/controller/directvolumemigration/migrate.go:40\ngithub.com/konveyor/mig-controller/pkg/controller/directvolumemigration.(*ReconcileDirectVolumeMigration).Reconcile()\n\t/Users/jmatthews/git/jwmatthews/mig-controller/pkg/controller/directvolumemigration/directvolumemigration_controller.go:126\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem()\n\t/Users/jmatthews/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.11/pkg/internal/controller/controller.go:215\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()\n\t/Users/jmatthews/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.11/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1()\n\t/Users/jmatthews/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20181127025237-2b1284ed4c93/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil()\n\t/Users/jmatthews/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20181127025237-2b1284ed4c93/pkg/util/wait/wait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until()\n\t/Users/jmatthews/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20181127025237-2b1284ed4c93/pkg/util/wait/wait.go:88\nruntime.goexit()\n\t/usr/local/Cellar/go/1.15.6/libexec/src/runtime/asm_amd64.s:1374"}
```

From the DirectMigrationVolume CR
```
status:
  conditions:
  - category: Required
    lastTransitionTime: "2021-01-23T21:42:13Z"
    message: Direct migration is ready
    status: "True"
    type: Ready
  - category: Advisory
    durable: true
    lastTransitionTime: "2021-01-23T21:44:07Z"
    message: 'The migration has failed.  See: Errors.'
    reason: CreateRsyncTransferPods
    status: "True"
    type: Failed
  errors:
  - ConfigMap "migration-controller" not found
  itinerary: VolumeMigration
  observedDigest: da14888484aa3d78010a846f6d62db4fe821f91abd42b239e87d885aaadb748e
  phase: CreateRsyncTransferPods
  phaseDescription: Creating Rsync daemon pods on the target cluster
  startTimestamp: "2021-01-23T21:42:13Z"
```

The 'MigMigration'
```
status:
  conditions:
  - category: Advisory
    lastTransitionTime: "2021-01-23T21:42:31Z"
    message: 'Step: 36/47'
    reason: WaitForDirectVolumeMigrationToComplete
    status: "True"
    type: Running
  - category: Required
    lastTransitionTime: "2021-01-23T21:36:34Z"
    message: The migration is ready.
    status: "True"
    type: Ready
  itinerary: Final
  observedDigest: 5796dad8ad3d97745c9c1ab450e152e4746f15a5f4acaa5c3553d32c5d422a7c
  phase: WaitForDirectVolumeMigrationToComplete
  pipeline:
  - completed: "2021-01-23T21:40:12Z"
    message: Completed
    name: Prepare
    started: "2021-01-23T21:36:30Z"
  - completed: "2021-01-23T21:41:53Z"
    message: Completed
    name: Backup
    progress:
    - 'Backup openshift-migration/0e44ae00-5dc3-11eb-9ca8-df7e5254778b-2d8ql: 99 out
      of estimated total of 99 objects backed up (19s)'
    started: "2021-01-23T21:40:12Z"
  - completed: "2021-01-23T21:42:26Z"
    message: Completed
    name: StageBackup
    started: "2021-01-23T21:41:53Z"
  - message: Skipped
    name: StageRestore
    skipped: true
  - completed: "2021-01-23T21:42:31Z"
    message: Waiting for Direct Image Migration to complete.
    name: DirectImage
    phase: WaitForDirectImageMigrationToComplete
    progress:
    - 0 total ImageStreams; 0 running; 0 successful; 0 failed
    started: "2021-01-23T21:42:26Z"
  - message: Creating Rsync daemon pods on the target cluster
    name: DirectVolume
    phase: WaitForDirectVolumeMigrationToComplete
    started: "2021-01-23T21:42:31Z"
  - message: Not started
    name: Restore
  - message: Not started
    name: Cleanup
  startTimestamp: "2021-01-23T21:36:30Z"
```

**Comment by eriknelson** [comment]
We're thinking this is probably due to the fact that John was running the controller via the HACKING.md instructions that disable the mig-controller. The migration-controller configmap is coupled with the overall template in the operator, so it makes sense that it would be *absent* if switched off via MigrationController CR.

I just ran through the upgrade steps seen here: https://docs.openshift.com/container-platform/4.6/migration/migrating_3_4/upgrading-migration-tool-3-4.html#migration-upgrading-migration-tool-4_migrating-3-4

I've encountered no problems and seen a 35-40% improvement in wall clock migration time with a direct migration of the sock-shop app compared to the 1.3.2 indirect migration.
